gammapy/time/lightcurve.py

```diff
@@ -112,3 +116,249 @@ def compute_fvar(self):
+                    livetime_to_add = (interval_tmax - obs_start).to('s')
+                # obs included in interval
+                elif interval_tmin <= obs_start and interval_tmax >= obs_stop:
+                    livetime_to_add = (obs_stop - obs_start).to('s')
+                else:
+                    pass
+
+                # Take into account dead time
+                livetime_to_add *= (1. - obs.observation_dead_time_fraction)
+                
+                # Compute excess
+                obs_measured_excess = obs_on - spec.alpha * obs_off
+
+                # Compute expected excess
+                # We use the effective livetime and the right energy threshold
+                e_idx = np.where(np.logical_and(spec.e_reco>=spec.lo_threshold,
+                                                spec.e_reco<=spec.hi_threshold))
+                counts_predicted_excess = calculate_predicted_counts(livetime=livetime_to_add,
+                                                                     aeff=spec.aeff,
+                                                                     edisp=spec.edisp,
+                                                                     model=spectral_model,
+                                                                     e_reco=spec.e_reco[e_idx])
+                obs_predicted_excess = np.sum(counts_predicted_excess.data.data)
+                obs_predicted_excess /= obs.observation_live_time_duration.to('s').value
+                obs_predicted_excess *= livetime_to_add.value
+
+                # compute effective normalisation between ON/OFF (for the interval)
+                interval_livetime += livetime_to_add
+                interval_alpha_mean += spec.alpha * obs_off
+                interval_alpha_mean_backup += spec.alpha * livetime_to_add
+                interval_measured_excess += obs_measured_excess
+                interval_predicted_excess += obs_predicted_excess
+                interval_on += obs_on
+                interval_off += obs_off
+                
+            # Fill time interval information
+            int_flux = spectral_model.integral(energy_range[0], energy_range[1])
+            if interval_off > 0.:
+                interval_alpha_mean /= interval_off
+            if interval_livetime > 0.:
+                interval_alpha_mean_backup /= interval_livetime
+
+            if interval_alpha_mean == 0.:  # use backup if necessary
+                interval_alpha_mean = interval_alpha_mean_backup
+
+            interval_flux = interval_measured_excess / interval_predicted_excess.value
+            interval_flux *= int_flux
+            interval_flux_error = int_flux / interval_measured_excess
+            # Gaussian errors, ToDo: should be improved
+            interval_flux_error *= np.sqrt(interval_on + interval_alpha_mean**2 * interval_off)
+
+            print('### start:{}, stop:{}'.format(interval_tmin, interval_tmax))
```

Comment:
No printing from Gammapy allowed.

Either just remove it, or, if you think it will be useful to people, then change this class to the following pattern of storing the relevant info and exposing a print_report or print_summary method.

```
est = LightCurveEstimator(...)
est.run(...) # what you call `lightcurve` now
est.lightcurve # The main result, stored on the estimator class for access after it runs
est.print_report() # print out summary or debugging info after the algorithm runs
```